### PR TITLE
Smoke test: Add test of a sequence of hdf5 commands

### LINKED ISSUE #17680: integrated hdf5 smoke test
Create an additional `hdf5` smoke test, per becker33 in https://github.com/spack/spack/pull/17360#pullrequestreview-450292260, that performs the following:

- Use well-structured `hdf5` file
- Check output of `h5dump` on the file
- Use `h5copy` to copy the file
- Ensure `h5diff` of file and copy are identical

### PR DESCRIPTION
**Pull request by tldahlgren**
Fixes #17680 

This PR adds a simple example-based test that takes an `hdf5` file through the dump, copy, and diff commands.

This PR has been tested against the following version(s):  1.8.10, 1.8.16, 1.10.1, 1.10.6, and 1.12.0.

TODO:
- [x] Install and run the tests against additional versions of the software

### CODE REVIEW

**Comment by adamjstewart** [comment]
@lrknox can you review?

File: var/spack/repos/builtin/packages/hdf5/package.py [hunk 1]
```diff
@@ -387,10 +389,36 @@ def _test_check_versions(self):
             self.run_test(exe, [option], spec_vers_str, None, installed=True,
                           purpose=reason, skip_missing=True)
 
+    def _test_example(self):
+        """This test performs copy, dump, and diff on an example hdf5 file."""
+        curr_dir = os.getcwd()
+        h5_file = os.path.join(curr_dir, 'data', 'spack.h5')
```
Comment (becker33) [inline var/spack/repos/builtin/packages/hdf5/package.py]:
```suggestion
        curr_dir = os.getcwd()
        h5_file = os.path.join(self.test_dir, 'data', 'spack.h5')
```

File: var/spack/repos/builtin/packages/hdf5/package.py [hunk 2]
```diff
@@ -387,10 +389,36 @@ def _test_check_versions(self):
...
+
+        reason = 'test: ensure h5dump produces expected output'
+        dump_file = os.path.join(curr_dir, 'data', 'dump.out')
```
Comment (becker33) [inline var/spack/repos/builtin/packages/hdf5/package.py]:
```suggestion
        dump_file = os.path.join(self.test_dir, 'data', 'dump.out')
```

**Review by becker33 — CHANGES_REQUESTED** [review]
Mostly LGTM. A couple changes to use `self.test_dir` instead of the cwd.

File: var/spack/repos/builtin/packages/hdf5/package.py (hunk 1 above)
Comment (tldahlgren) [reply var/spack/repos/builtin/packages/hdf5/package.py]: Also need to remove `curr_dir` then.

**Pull request merged**